synapse/static/client/login/js/login.js

```diff
@@ -7,26 +7,19 @@ window.matrixLogin = {
 var title_pre_auth = "Log in with one of the following methods";
 var title_post_auth = "Logging in...";
 
-var submitPassword = function(user, pwd) {
-    console.log("Logging in with password...");
+var submitLogin = function(type, data) {
+    console.log("Logging in with " + type);
     set_title(title_post_auth);
-    var data = {
-        type: "m.login.password",
-        user: user,
-        password: pwd,
-    };
-    $.post(matrixLogin.endpoint, JSON.stringify(data), function(response) {
-        matrixLogin.onLogin(response);
-    }).fail(errorFunc);
-};
 
-var submitToken = function(loginToken) {
-    console.log("Logging in with login token...");
-    set_title(title_post_auth);
-    var data = {
-        type: "m.login.token",
-        token: loginToken
-    };
+    // Add the login type.
+    data.type = type;
```

Comment:
This modifies the input value which is kind of meh, but I think this code is contained enough that it is OK + we control the values being sent to this function.